pkg/ingester/ingester_v2.go

```diff
@@ -110,6 +112,37 @@ func (u *userTSDB) PostDeletion(metrics ...labels.Labels) {
 	}
 }
 
+// blocksToDelete filters the input blocks and returns the blocks which are safe to be deleted from the ingester.
+func (u *userTSDB) blocksToDelete(blocks []*tsdb.Block) map[ulid.ULID]struct{} {
+	if u.DB == nil {
+		return map[ulid.ULID]struct{}{}
```

Comment:
Nit:

```suggestion
		return nil
```

Reply:
(Same below)

Reply:
I thought `nil` maps would cause panics, but apparently it does not

Reply:
`nil` maps are fine for reading, iterating and deleting, but not for writing new elements into them.